### PR TITLE
Watch Folders

### PR DESCRIPTION
**Pull request by rushitote**
## Description

https://github.com/ente-io/bada-frame/pull/583

### CODE REVIEW

File: src/utils/ipcComms.ts [hunk 1]
```diff
@@ -64,6 +66,27 @@ export default function setupIpcComs(
         return files;
     });
 
+    ipcMain.handle('select-folder', async () => {
```
Comment (abhinavkgrd) [inline src/utils/ipcComms.ts:69]: duplicate of selet-dir

File: src/utils/watch.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+export async function isFolderExists(dirPath: string) {
+    return await fs
+        .stat(dirPath)
+        .then((stats) => {
+            return stats.isDirectory();
+        })
+        .catch(() => false);
+}
+
+export function initWatcher(mainWindow: BrowserWindow) {
```
Comment (abhinavkgrd) [inline src/utils/watch.ts:85]: move to a chowkidar util

File: src/preload.ts [hunk 3]
```diff
@@ -10,6 +10,17 @@ import {
...
+    initWatcher,
+    registerWatcherFunctions,
+    setWatchMappings,
+    addWatchMapping,
+    removeWatchMapping,
+    isFolderExists,
+} from './utils/watch';
+import path from 'path';
 
 const { ipcRenderer } = electron;
```
Comment (abhinavkgrd) [inline src/preload.ts:26]: create a seperate util for export functions

Comment (abhinavkgrd) [reply src/preload.ts:26]: add stats exists check for getExportRecord

File: src/api/watch.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        WatchServiceInstance: any,
+        folderPath: string
+    ) => Promise<void>
+) {
+    ipcRenderer.removeAllListeners('watch-add');
+    ipcRenderer.removeAllListeners('watch-change');
+    ipcRenderer.removeAllListeners('watch-unlink');
+    ipcRenderer.on('watch-add', async (_, filePath: string) => {
+        filePath = filePath.split(path.sep).join(path.posix.sep);
+        await addFile(WatchServiceInstance, await getElectronFile(filePath));
```
Comment (abhinavkgrd) [inline src/api/watch.ts:102]: we should send the files to bada-frame here and let bada-frame execute the function.
executing its functions from here doesn't seem right

**Review by ua741 — APPROVED** [review]

**Review by ua741 — DISMISSED** [review]

File: .gitmodules [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [submodule "bada-frame"]
 	path = ui
 	url = https://github.com/ente-io/bada-frame
-	branch = main
+	branch = watch
```
Comment (ua741) [inline .gitmodules:4]: LGTM. As discussed, can merge when we switch to main

**Comment by abhinavkgrd** [comment]
dont merge manav bhaiya have asked a small change... will include that too 

https://github.com/ente-io/bada-frame/pull/699

**Review by mnvr — APPROVED** [review]

**Pull request merged**